Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -60,311 +369,258 @@ struct MacroSystem {
+        if let lastNewline = leadingTrivia.pieces.lastIndex(where: { $0.isNewline }),
+          leadingTrivia.pieces[lastNewline...].allSatisfy(\.isWhitespace),
+          node.trailingTrivia.isEmpty,
+          node.nextToken(viewMode: .sourceAccurate)?.leadingTrivia.first?.isNewline ?? false
+        {
+          // If the attribute is on its own line based on the following conditions,
+          // remove the newline from it so we don’t end up with an empty line
+          //  - Trailing trivia ends with a newline followed by arbitrary number of spaces or tabs
+          //  - There is no trailing trivia and the next token starts on a new line
+          leadingTrivia = Trivia(pieces: leadingTrivia.pieces[..<lastNewline])
+        }
+        // Drop any spaces or tabs from the trailing trivia because there’s no
+        // more attribute they need to separate.
+        let trailingTrivia = Trivia(pieces: attribute.trailingTrivia.drop(while: { $0.isSpaceOrTab }))
+        triviaToAttachToNextToken += leadingTrivia + trailingTrivia
+      } else {
+        filteredAttributes.append(.attribute(attribute))
+      }
+    }
+    return AttributeListSyntax(filteredAttributes)
+  }
+
+  override func visit(_ token: TokenSyntax) -> TokenSyntax {
+    if !triviaToAttachToNextToken.isEmpty {
+      defer { triviaToAttachToNextToken = Trivia() }
+      return token.with(\.leadingTrivia, triviaToAttachToNextToken + token.leadingTrivia)
+    } else {
+      return token
+    }
+  }
+}
+
+private let diagnosticDomain: String = "SwiftSyntaxMacroExpansion"
+
+private enum MacroApplicationError: DiagnosticMessage, Error {
+  case accessorMacroOnVariableWithMultipleBindings
+  case malformedAccessor
+
+  var diagnosticID: MessageID {
+    return MessageID(domain: diagnosticDomain, id: "\(self)")
+  }
+
+  var severity: DiagnosticSeverity { return .error }
+
+  var message: String {
+    switch self {
+    case .accessorMacroOnVariableWithMultipleBindings:
+      return """
+        swift-syntax applies macros syntactically but there is no way to represent a variable \
+        declaration with multiple bindings that have accessors syntactically. \
+        While the compiler allows this expansion, swift-syntax cannot represent it and thus \
+        disallows it.
+        """
+    case .malformedAccessor:
+      return """
+        Macro returned a mal-formed accessor. Accessors should start with an introducer like 'get' or 'set'.
+        """
+    }
+  }
+}
+
 /// Syntax rewriter that evaluates any macros encountered along the way.
-class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
+private class MacroApplication<Context: MacroExpansionContext>: SyntaxRewriter {
   let macroSystem: MacroSystem
   var context: Context
-  var skipNodes: Set<Syntax> = []
+  var indentationWidth: Trivia
+  /// Nodes that we have already handled in `visitAny` and that should be visited
+  /// in the node-specific handling function.
```

Comment:
No, that’s not correct. The nodes will be in `skipVisitAnyHandling` if the node-specific visit should be called instead of `visitAny`. What do you think about the following?

```swift
  /// Nodes that we have already handled in `visitAny`. If they are visited again, `visitAny` should return `nil` so that the node-specific visit method is called.

Reply:
Maybe just switch to present tense = "Nodes that we are currently handling"